staging/src/k8s.io/client-go/util/workqueue/queue_test.go

```diff
@@ -167,6 +167,28 @@ func TestLen(t *testing.T) {
 	}
 }
 
+// TestIsQueued ensures that we can check the active queue for specific objects.
+func TestIsQueued(t *testing.T) {
+	q := workqueue.New()
+
+	q.Add("foo")
+	q.Add("bar")
+	if !q.IsQueued("foo") {
+		t.Errorf("Expected foo to be reported as queued")
+	}
+	if !q.IsQueued("bar") {
+		t.Errorf("Expected bar to be reported as queued")
+	}
+	if q.IsQueued("baz") {
+		t.Errorf("baz should not be reported as queued")
+	}
+	_, _ = q.Get()
+	q.Done("foo")
+	if q.IsQueued("foo") {
```

Comment:
Can you actually move this up one line, before the Done()? I think as soon as Get() has returned something it should no longer be considered queued (otherwise a caller checking IsQueued may fail to add something to the queue even though they should).

Reply:
Have done! :)